src/compiler/checker.ts

```diff
@@ -31297,18 +31298,16 @@ namespace ts {
                         if (!hasParseDiagnostics(sourceFile)) {
                             const span = getSpanOfTokenAtPosition(sourceFile, node.pos);
                             const diagnostic = createFileDiagnostic(sourceFile, span.start, span.length, Diagnostics.await_expressions_are_only_allowed_within_async_functions_and_at_the_top_levels_of_modules);
-                            const func = getContainingFunction(node);
-                            if (func && func.kind !== SyntaxKind.Constructor && (getFunctionFlags(func) & FunctionFlags.Async) === 0) {
-                                const relatedInfo = createDiagnosticForNode(func, Diagnostics.Did_you_mean_to_mark_this_function_as_async);
+                            if (containingFunctionOrClassStaticBlock && containingFunctionOrClassStaticBlock.kind !== SyntaxKind.ClassStaticBlockDeclaration && containingFunctionOrClassStaticBlock.kind !== SyntaxKind.Constructor && (getFunctionFlags(containingFunctionOrClassStaticBlock) & FunctionFlags.Async) === 0) {
+                                const relatedInfo = createDiagnosticForNode(containingFunctionOrClassStaticBlock, Diagnostics.Did_you_mean_to_mark_this_function_as_async);
                                 addRelatedInfo(diagnostic, relatedInfo);
                             }
                             diagnostics.add(diagnostic);
                         }
                     }
                 }
-
-                if (node.flags & NodeFlags.ClassStaticBlockContext) {
-                    error(node, Diagnostics.Await_expression_cannot_be_used_inside_class_static_block);
+                else if (containingFunctionOrClassStaticBlock && isClassStaticBlockDeclaration(containingFunctionOrClassStaticBlock)) {

```

Comment:
If you make this the first branch of the `if` statement, then you shouldn't need to exclude `containingFunctionOrClassStaticBlock.kind !== SyntaxKind.ClassStaticBlockDeclaration` on line 31301.